subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -946,21 +946,23 @@ static uint8_t ticker_resolve_collision(struct ticker_node *nodes,
 			 * the ticks_slot_window.
 			 */
 			uint8_t next_not_ticks_slot_window =
-					(!TICKER_HAS_SLOT_WINDOW(ticker_next) ||
-					 ((acc_ticks_to_expire +
-					   ticker_next->ext_data->ticks_slot_window -
-					   ticker_next->ticks_slot) <
-					  ticker->ticks_slot));
+					!TICKER_HAS_SLOT_WINDOW(ticker_next) ||
+					(ticker_next->ext_data->is_drift_in_window &&
+					 TICKER_HAS_SLOT_WINDOW(ticker)) ||
+					((acc_ticks_to_expire +
+					  ticker_next->ext_data->ticks_slot_window -
+					  ticker_next->ticks_slot) <
+					 ticker->ticks_slot);
 
 			/* Can the current ticker with ticks_slot_window be
 			 * scheduled after the colliding ticker?
 			 */
 			uint8_t curr_has_ticks_slot_window =
-					(TICKER_HAS_SLOT_WINDOW(ticker) &&
-					 ((acc_ticks_to_expire +
-					   ticker_next->ticks_slot) <
-					  (ticker->ext_data->ticks_slot_window -
-					   ticker->ticks_slot)));
+					TICKER_HAS_SLOT_WINDOW(ticker) &&
+					((acc_ticks_to_expire +
+					  ticker_next->ticks_slot) <=
+					 (ticker->ext_data->ticks_slot_window -
+					  ticker->ticks_slot));
```

Comment:
Several of these calculations are getting quite complex. Should functions be implemented to help calculate `next_not_ticks_slot_window` and `curr_has_ticks_slot_window`?

The worst part of these is that the calculations are undocumented: It's not clear to me why they are calculated as such, and are hard to maintain.

Reply:
Updated the rules for collision resolution which are documented in the function description.

The logical expression at line 1012-1019 reflect the documented rules, the variables are accordingly named to reflect the rule sentences.